tests/ops/test_channel_ops.py

```diff
@@ -137,6 +126,24 @@ def test_p_arbitrary(self, p, tol):
         expected_K1 = np.sqrt(p) * X
         assert np.allclose(op(p, wires=0).kraus_matrices[1], expected_K1, atol=tol, rtol=0)
 
+    @pytest.mark.parametrize("angle", np.linspace(0, 2 * np.pi, 7))
+    def test_grad_bitflip(self, angle, tol):
+        """Test that analytical gradient is computed correctly for different states. Channel
+        grad recipes are independent of channel parameter"""
+
+        dev = qml.device("default.mixed", wires=1)
+        prob = 0.5
+
+        @qml.qnode(dev)
+        def circuit(p):
+            qml.RX(angle, wires=0)
+            qml.BitFlip(p, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        gradient = np.squeeze(qml.grad(circuit)(prob))
+        assert gradient == circuit(1)-circuit(0)
+        assert np.allclose(gradient, (-2*np.cos(angle)))
```

Comment:
Thanks for adding these @ixfoduap 💪 The end function is deceptively simple, but I know from previous experience that to get there it can be a bit of algebra!